File: pset1/mario.c
//creating a pattern for mario
/*
    #  #
   ##  ##
  ###  ###
 ####  ####
*/
#include <stdio.h>
#include <cs50.h>

void mario(int height);

int main()
{  
    int height = 0;
    do
    {
        height = get_int("Enter hight:\n");
    }
    while (height < 1 || height > 8);
    
    mario(height);
}



void mario(int height)
{
    for (int i = 0 ; i < height ; i++)        // For hight 
    {   
        for (int j = 0 ; j < height ; j++)    // loop for the first set of #
        {
            if (j < (height - 1 - i))
            {
                printf(" ");
            }
            else 
            {
                printf("#");
            }
        }
        
        printf("  ");
        for (int j = 0 ; j <= i ; j++)    // loop for the second set
        {
            printf("#"); 
        }
        printf("\n");
    }
}

File: pset6/mario/mario.py
from cs50 import get_int


def main():
    while True:
        height = get_int("enter the height: ")
        if height in range(1, 9):
            break
        
    mario(height)


def mario(n):
    i = 0
    for i in range(n):
        print(" " * ((n - i) - 1), end="")
        print("#" * (i + 1), end="")
        print("  ", end="")
        print("#" * (i + 1), end="")
        print()
        

main()
